### PR TITLE
fix: MenuButton is styled correctly when aria-expanded is a string

### LINKED ISSUE #31804: [Bug]: MenuButton transparent appearance styling not consistent when rendered within a Popover trigger element
### Library

React Northstar / v0 (@fluentui/react-northstar)

### System Info

```shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://stackblitz.com/edit/ixwcan-uutrkp?file=src%2Fexample.tsx

## Actual Behavior
Button color on rest state is blue for transparent state
https://stackblitz.com/edit/ixwcan-uutrkp?file=src%2Fexample.tsx

## Expected Behavior
Button color on rest state should be black for the transparent state
https://react.fluentui.dev/iframe.html?viewMode=docs&id=components-button-menubutton--default#appearance
image

Stackblitz Repro - 
https://stackblitz.com/edit/ixwcan-uutrkp?file=src%2Fexample.tsx

### Logs

_No response_

### Requested priority

Low

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by smhigley**
## Previous Behavior

When `MenuButton` was passed in `aria-expanded` as a string, the `useMenuButtonStyles` would incorrectly apply the expanded styles even when the value was `'false'`.

## New Behavior

`useMenuButton` normalizes the value to always be a boolean.

## Related Issue(s)

- Fixes #31804

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31809/perf-test-react-components/Avatar-mount_base.html">604</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31809/perf-test-react-components/Avatar-mount.html">649</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31809/perf-test-react-components/Button-mount_base.html">298</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31809/perf-test-react-components/Button-mount.html">302</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31809/perf-test-react-components/Field-mount_base.html">1129</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31809/perf-test-react-components/Field-mount.html">1108</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31809/perf-test-react-components/FluentProvider-mount_base.html">713</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31809/perf-test-react-components/FluentProvider-mount.html">699</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31809/perf-test-react-components/FluentProviderWithTheme-mount_base.html">72</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31809/perf-test-react-components/FluentProviderWithTheme-mount.html">81</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31809/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">31</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31809/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">32</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31809/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">73</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31809/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">77</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31809/perf-test-react-components/MakeStyles-mount_base.html">854</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31809/perf-test-react-components/MakeStyles-mount.html">854</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31809/perf-test-react-components/Persona-mount_base.html">1770</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31809/perf-test-react-components/Persona-mount.html">1690</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31809/perf-test-react-components/SpinButton-mount_base.html">1448</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31809/perf-test-react-components/SpinButton-mount.html">1371</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31809/perf-test-react-components/SwatchPicker-mount_base.html">1626</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/31809/perf-test-react-components/SwatchPicker-mount.html">1610</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr> | `41.886 kB`<br />`12.133 kB` | `41.912 kB`<br />`12.137 kB` | `26 B` <br />`4 B` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr> | `49.898 kB`<br />`13.725 kB` | `49.924 kB`<br />`13.73 kB` | `26 B` <br />`5 B` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ReactComponents.fixture.js'>react-components: entire library</abbr> | `1.094 MB`<br />`270.62 kB` | `1.094 MB`<br />`270.624 kB` | `26 B` <br />`4 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-breadcrumb</samp> <br /> <abbr title='bundle-size/index.fixture.js'>@fluentui/react-breadcrumb - package</abbr> | `113.657 kB`<br />`31.532 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr> | `37.105 kB`<br />`10.785 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr> | `43.516 kB`<br />`12.076 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr> | `53.035 kB`<br />`12.546 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr> | `100.845 kB`<br />`28.873 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr> | `93.629 kB`<br />`27.077 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr> | `14.356 kB`<br />`5.798 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr> | `16.879 kB`<br />`6.664 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr> | `14.42 kB`<br />`5.934 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `69.141 kB`<br />`20.157 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `211.741 kB`<br />`60.957 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `44.442 kB`<br />`14.607 kB` |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr> | `98.898 kB`<br />`29.801 kB` |
| <samp>react-message-bar</samp> <br /> <abbr title='bundle-size/Default.fixture.js'>MessageBar (all components)</abbr> | `24.596 kB`<br />`9.18 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `8.39 kB`<br />`2.64 kB` |
| <samp>react-tag-picker</samp> <br /> <abbr title='bundle-size/Package.fixture.js'>@fluentui/react-tag-picker - package</abbr> | `181.136 kB`<br />`54.463 kB` |
| <samp>react-timepicker-compat</samp> <br /> <abbr title='bundle-size/TimePicker.fixture.js'>TimePicker</abbr> | `104.384 kB`<br />`34.782 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/54f5ccc5635d26d96ed43b005927a0eebdb79b94'>54f5ccc5635d26d96ed43b005927a0eebdb79b94</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-button-9d0e8c6a-5acc-4b52-a96f-57d04ac83672.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-button-9d0e8c6a-5acc-4b52-a96f-57d04ac83672.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=31809&commitId=bc6d94d695089950323adefa03946ce871c89917&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.badgeMask.normal.chromium.png|5| Changed|
|Avatar Converged.badgeMask - RTL.normal.chromium.png|3| Changed|

</details>


  <details><summary><b>Image Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Image Converged.Image Variations Fallback.normal.chromium.png|0| Added|

</details>


  <details><summary><b>SwatchPicker Converged</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|SwatchPicker Converged.spacing.default.chromium.png|0| Added|
|SwatchPicker Converged.Shape.default.chromium.png|0| Removed|
|SwatchPicker Converged.default - High Contrast.default.chromium.png|0| Removed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Review by khmakoto — APPROVED** [review]

File: change/@fluentui-react-button-9d0e8c6a-5acc-4b52-a96f-57d04ac83672.json (hunk 1 above)
Comment (smhigley) [reply change/@fluentui-react-button-9d0e8c6a-5acc-4b52-a96f-57d04ac83672.json:1]: unrelated to the PR

**Pull request merged**